core/src/test/java/de/unistuttgart/iaas/amyassist/amy/core/configuration/EnvironmentConfigurationLoaderTest.java

```diff
@@ -0,0 +1,319 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information see notice.md
+ */
+
+package de.unistuttgart.iaas.amyassist.amy.core.configuration;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Reference;
+import de.unistuttgart.iaas.amyassist.amy.core.io.Environment;
+import de.unistuttgart.iaas.amyassist.amy.test.FrameworkExtension;
+import de.unistuttgart.iaas.amyassist.amy.test.TestFramework;
+
+/**
+ * Test class for {@link EnvironmentConfigurationLoader}
+ * 
+ * @author Tim Neumann
+ */
+@ExtendWith(FrameworkExtension.class)
+class EnvironmentConfigurationLoaderTest {
+
+	private static final int NUM_RUNS = 1000;
+	private static final int MAX_LENGTH = 10;
+
+	@Reference
+	private TestFramework testFramework;
+
+	private EnvironmentConfigurationLoader loader;
+
+	private Map<String, String> envVars;
+
+	private static Random rng;
+	private static long seed;
+
+	/**
+	 * Before all
+	 */
+	@BeforeAll
+	static void init() {
+		seed = (long) (Math.random() * Integer.MAX_VALUE);
+		EnvironmentConfigurationLoaderTest.rng = new Random(seed);
+	}
+
+	/**
+	 * Setup before each test
+	 */
+	@BeforeEach
+	void setup() {
+
+		this.envVars = new HashMap<>();
+		Environment env = this.testFramework.mockService(Environment.class);
+		Mockito.when(env.getEnvironmentVariables()).thenReturn(this.envVars);
+
+		this.loader = this.testFramework.setServiceUnderTest(EnvironmentConfigurationLoaderImpl.class);
+	}
+
+	/**
+	 * Tests just the trivial behavior.
+	 * <p>
+	 * Test method for
+	 * {@link de.unistuttgart.iaas.amyassist.amy.core.configuration.EnvironmentConfigurationLoaderImpl#load(java.lang.String, java.util.Properties)}.
+	 */
+	@Test
+	void testLoadStandard() {
+		this.envVars.put("config1_key1", "value11");
+		this.envVars.put("config1_key2", "value12");
+		this.envVars.put("config2_key1", "value21");
+
+		Properties orig1 = new Properties();
+		Properties orig2 = new Properties();
+
+		orig1.setProperty("key1", "orig11");
+		orig1.setProperty("key2", "orig12");
+		orig2.setProperty("key1", "orig21");
+
+		Properties res1 = this.loader.load("config1", orig1);
+		Properties res2 = this.loader.load("config2", orig2);
```

Comment:
splitt this in two test cases

Reply:
No. This method is testing the standard uses of the load method. And these are two very standard uses of that method.

Reply:
you don't have to test two Properties files in the same equivalence class